src/test/ui/suggestions/field-access.stderr

```diff
@@ -47,6 +48,20 @@ help: you might have meant to use field `b` of type `B`
 LL |     match a.b {
    |           ^^^
 
-error: aborting due to 3 previous errors
+error[E0308]: mismatched types
+  --> $DIR/field-access.rs:32:9
+   |
+LL |     match foo {
+   |           --- this expression has type `Foo`
+LL |
+LL |         1u32 => (),
+   |         ^^^^ expected union `Foo`, found `u32`
+   |
+help: you might have meant to use field `bar` of type `u32`
```

Comment:
This message is slightly inaccurate because it doesn't quite account for how unions are represented internally. Would you be up to fix it in this PR?

Reply:
I'd like to get some elaboration on this as I don't use unions very often. Is it something about the wording?

Reply:
"you might have meant to use field `bar` of type `u32`" implies that `u32` has a field called `bar`. What is actually happening is that there's a `union Foo` that  has a field `bar` of type `u32`.

Reply:
We can just create a ticket to handle that rewording separately.

Reply:
Opened #81655.